src/Preg.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+/**
+ * This class replaces preg_* functions to better handle UTF8 strings.
```

Comment:
What is that "better" ? please describe playing with `/u`, especially:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3446#discussion_r172073746
(so one would know shall he pass `/u`, shall he not, or all works)